crypto/x509v3/v3_alt.c

```diff
@@ -143,9 +153,16 @@ int GENERAL_NAME_print(BIO *out, GENERAL_NAME *gen)
     int i;
     switch (gen->type) {
     case GEN_OTHERNAME:
-        BIO_printf(out, "othername:<unsupported>");
+    {
+        ASN1_TYPE *value = gen->d.otherName->value;
+        int nid = OBJ_obj2nid(gen->d.otherName->type_id);
+        if ((nid == NID_id_on_dnsSRV) && (value->type == V_ASN1_IA5STRING)) {
+            BIO_printf(out, "SRVName:%s", value->value.ia5string->data);
+        } else {
+            BIO_printf(out, "othername:<unsupported>");
+        }
```

Comment:
Wouldn't it be more appropriate to provide more generalized solution? I mean instead of checking specifically for NID_id_on_dnsSRV, one could check for nid being just sane and pull name with OBJ_nid2sn? Then one can of course check even for type being suitable for printing it as string...

Reply:
I'm not sure if a generalized solution is feasible. Othername entries are OID + arbitrary ASN.1 structure. SRVName just happens to be a very simple ASN.1 basic type that can be easily converted to a string. For example ``id-pkinit-san`` (Kerberos principal name for public key kinit) is a nested ASN.1 SEQUENCE.

I could provide a slightly more general solution for ``type == V_ASN1_IA5STRING`` and ``nid != NID_undef``. But it would print ``id-on-dnsSRV:_https.www.example.org``, which looks ugly.

PS: You should be able to use ``otherName.1 = id-on-dnsSRV;IA5STRING:_https.www.example.org`` with my patch. I haven't tried it, though.

Reply:
> I'm not sure if a generalized solution is feasible.

I did write "one can of course check even for type being suitable for printing as string". In other words suggestion is not generalized GENERALIZED, but generalized for "printable" OIDs, i.e. `if (nid != NID_undef && value->type == V_ASN1_IA5STRING)`, maybe even other strings...